Documentation/ceph-cluster-crd.md

```diff
@@ -106,6 +101,28 @@ If a node does not specify any configuration then it will inherit the cluster le
 - `config`: Config settings applied to all OSDs on the node unless overridden by `devices` or `directories`. See the [config settings](#osd-configuration-settings) below.
 - [storage selection settings](#storage-selection-settings)
 
+When `useAllNodes` is set to `true`, Rook attempts to make Ceph cluster management as hands-off as
+possible, taking steps -- sometimes aggressive steps -- to ensure the cluster is healthy and usable.
+If a Kubernetes node becomes no longer `Ready`, Rook will begin removing that node from the Ceph
+cluster. This may be due to a node failure, a network partition, or a node being cordoned for
```

Comment:
Cordoning a node was one of the cases where we don't want to automatically remove the OSDs. From the original issue #2253 it was said: "If I temporary cordon a node to prevent user jobs from running in it and operator restarts, it will delete all OSDs from the node. I'd definitely not want that." 

In other words, if a node is in maintenance, we don't want to automatically remove OSDs from that node. Even though the OSDs are down for some (short) time, it is expected that they start up again as soon as the node is started again.

Thus, I'd argue that a node's schedulability only affects whether to add OSDs, but does **not** affect whether to remove OSDs.

Reply:
The problem here is that with `useAllNodes: true` there is no way to remove nodes unless we allow Rook to be aggressive about it like this. The PR to wait until OSDs are safe to remove should help this somewhat, as it will likely take longer than node maintenance times to rebalance a Ceph cluster.

Reply:
Updated but still need to alter docs and commit message.